### PR TITLE
Up-to-date Swagger

### PR DESCRIPTION
**Pull request by maxkfranz**
- Add `SWAGGER_HOST` env var.  This needs to be set for Docker configs etc.  By default, it points to `localhost:3000` so that it works out of the box for people who just clone the repo and start the service.  @jvwong, we may need a corresponding update to the Docker compose configs for Biofactoid.
- Docs for `/search`, `/get`, and `/map`.  @jvwong, could you check that your `/map` route isn't missing anything here?
- Examples are included with the Swagger docs, so on the page the user can just click a button to run an example query.  They don't have to write a bunch of JSON just to test things out.
- The Swagger docs page automatically displays the version number specified in `package.json`, so new releases will always display properly without extra prep work.
- For `localhost:3000`, only `http` is allowed.  For others, the user can select `https` or `http` in a dropdown.  We use `https` by default for a public instance.
- More updates to the readme could be made re. #116 after #117 has been merged in.

@jvwong, could you double check that there's nothing missing here, esp. re. `/map`?  If you could try out the query UI in Swagger to see if there's anything I missed, I'd appreciate that.

@metincansiper, you originally set up the Swagger config.  Could you confirm that there's nothing mis-set by these changes?  Thanks

Main page:

Screen Shot 2021-08-11 at 12 15 27

Docs & example for `/search`:

Screen Shot 2021-08-11 at 12 15 25

Screen Shot 2021-08-11 at 12 22 13

Result for example query:

Screen Shot 2021-08-11 at 12 15 52

Ref. :

- #116
- #117

### CODE REVIEW

File: src/server/routes/index.js [hunk 1]
```diff
@@ -35,20 +52,86 @@ router.post('/search', function(req, res){
...
+ *         required: true
+ *         schema:
+ *           type: object
+ *           properties:
+ *             dbfrom:
+ *               type: string
+ *               description: The MIRIAM database prefix of the provided ID.
+ *               example: "ncbigene"
+ *               required: true
+ *             id:
```
Comment (jvwong) [inline src/server/routes/index.js:121]: This is correct - also  I accepted  an array of strings as well.

**Review by jvwong — APPROVED** [review]

**Comment by jvwong** [comment]
>     * Add `SWAGGER_HOST` env var.  This needs to be set for Docker configs etc.  By default, it points to `localhost:3000` so that it works out of the box for people who just clone the repo and start the service.  @jvwong, we may need a corresponding update to the Docker compose configs for Biofactoid.

I believe leaving the default `SWAGGER_HOST` is fine as is for any Docker instances.

**Comment by maxkfranz** [comment]
> > ```
> > * Add `SWAGGER_HOST` env var.  This needs to be set for Docker configs etc.  By default, it points to `localhost:3000` so that it works out of the box for people who just clone the repo and start the service.  @jvwong, we may need a corresponding update to the Docker compose configs for Biofactoid.
> > ```
> 
> 
> I believe leaving the default `SWAGGER_HOST` is fine as is for any Docker instances.

Would that work for `grounding.baderlab.org` when someone goes to the swagger docs at that URL?  A public site should not be able to access `localhost` from the (remote) client side.

**Comment by jvwong** [comment]
- internet traffic to `grounding.baderlab.org` gets re-rerouted to our development VM (devIP:`GROUNDING_SEARCH_PORT`)
- On the development VM, traffic to devIP:`GROUNDING_SEARCH_PORT` gets mapped to the grounding container

So I'm thinking that as long as the docker-config.yml specifies the right port inside the container, its ok?

**Comment by maxkfranz** [comment]
I think the swagger var is for the browser/client, so backend server conf may not affect things. I may be wrong. If you have it tested/working as-is with your docker images, great

**Review by metincansiper — APPROVED** [review]
@maxkfranz looks fine to me.

**Comment by jvwong** [comment]
> I think the swagger var is for the browser/client, so backend server conf may not affect things. I may be wrong. If you have it tested/working as-is with your docker images, great

Lemme run some more tests

**Comment by jvwong** [comment]
The corresponding Docker container  is working both  locally and on dev  @ https://grounding.baderlab.org/api/docs/

**Comment by jvwong** [comment]
Ah! b I see what you mean about the host on  the client...

**Comment by maxkfranz** [comment]
Sounds good. As long as grounding.baderlab.org has the swagger var working for it, then we can link to it in the readme

**Pull request merged**